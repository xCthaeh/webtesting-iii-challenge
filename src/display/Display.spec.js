// Test away!
import React from "react";
import renderer from "react-test-renderer";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/react/cleanup-after-each";

describe("<Display />", () => {
  it("should match snapshot", () => {
    const tree = renderer.create(<Display />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render", () => {
    render(<Display closed="true" locked="true" />);
  });

  it("should default to unlocked", () => {
    const { getByText } = render(<Display />);
    getByText(/unlocked/i);
  });

  it("should default to open", () => {
    const { getByText } = render(<Display />);
    getByText(/open/i);
  });

  it("should be locked", () => {
    const { getByText } = render(<Display closed={true} locked={true} />);
    getByText(/locked/i);
  });

  it("should be closed", () => {
    const { getByText } = render(<Display closed={true} locked={true} />);
    getByText(/closed/i);
  });

  it("should be unlocked", () => {
    const { getByText } = render(<Display closed={false} locked={false} />);
    getByText(/unlocked/i);
  });

  it("should be green-led", () => {
    const { getByText } = render(<Display closed={false} locked={false} />);
    expect(getByText(/open/i).classList[1]).toEqual("green-led");
    expect(getByText(/unlocked/i).classList[1]).toEqual("green-led");
  });

  it("should be red-led", () => {
    const { getByText } = render(<Display closed={true} locked={true} />);
    expect(getByText(/closed/i).classList[1]).toEqual("red-led");
    expect(getByText(/locked/i).classList[1]).toEqual("red-led");
  });
});
