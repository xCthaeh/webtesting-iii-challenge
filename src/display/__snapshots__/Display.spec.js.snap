// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Display /> should match snapshot 1`] = `
<div
  className="display panel"
>
  <div
    className="led green-led"
  >
    Unlocked
  </div>
  <div
    className="led green-led"
  >
    Open
  </div>
</div>
`;
